libs/server-sdk/src/data_sources/data_source_event_handler.cpp

```diff
@@ -0,0 +1,110 @@
+#include "data_source_event_handler.hpp"
+
+#include <launchdarkly/encoding/base_64.hpp>
+#include <launchdarkly/serialization/json_flag.hpp>
+#include <launchdarkly/serialization/json_primitives.hpp>
+#include <launchdarkly/serialization/json_rule_clause.hpp>
+#include <launchdarkly/serialization/json_sdk_data_set.hpp>
+#include <launchdarkly/serialization/json_segment.hpp>
+#include <launchdarkly/serialization/value_mapping.hpp>
+
+#include <boost/core/ignore_unused.hpp>
+#include <boost/json.hpp>
+#include <unordered_map>
+
+#include <utility>
+
+#include "tl/expected.hpp"
+
+namespace launchdarkly::server_side::data_sources {
+
+static char const* const kErrorParsingPut = "Could not parse PUT message";
+static char const* const kErrorPutInvalid =
+    "PUT message contained invalid data";
+static char const* const kErrorParsingPatch = "Could not parse PATCH message";
+static char const* const kErrorPatchInvalid =
+    "PATCH message contained invalid data";
+static char const* const kErrorParsingDelete = "Could not parse DELETE message";
+static char const* const kErrorDeleteInvalid =
+    "DELETE message contained invalid data\"";
+
+static tl::expected<DataSourceEventHandler::FlagPatch, JsonError> tag_invoke(
+    boost::json::value_to_tag<tl::expected<DataSourceEventHandler::FlagPatch,
+                                           JsonError>> const& unused,
+    boost::json::value const& json_value) {
+    // TODO: Implement
+    return tl::unexpected(JsonError::kSchemaFailure);
+}
+
+static tl::expected<DataSourceEventHandler::SegmentPatch, JsonError> tag_invoke(
+    boost::json::value_to_tag<tl::expected<DataSourceEventHandler::SegmentPatch,
+                                           JsonError>> const& unused,
+    boost::json::value const& json_value) {
+    // TODO: Implement
+    return tl::unexpected(JsonError::kSchemaFailure);
+}
+
+static tl::expected<DataSourceEventHandler::DeleteData, JsonError> tag_invoke(
+    boost::json::value_to_tag<tl::expected<DataSourceEventHandler::DeleteData,
+                                           JsonError>> const& unused,
+    boost::json::value const& json_value) {
+    boost::ignore_unused(unused);
```

Comment:
Will look at cleaning this up potentially when streaming is implemented. Currently it is a copy/paste.